torch/utils/data/datapipes/_typing.py

```diff
@@ -530,8 +530,10 @@ def wrap_generator(*args, **kwargs):
                 #       Part of https://github.com/pytorch/data/issues/284
                 datapipe = args[0]
                 msg = "thrown by __iter__ of"
-                full_msg = f"{msg} {datapipe.__class__.__name__}({_generate_input_args_string(datapipe)})"
-                if len(e.args) >= 1 and msg not in e.args[0]:
+                single_iterator_msg = "single iterator per IterDataPipe constraint"
+                has_len = hasattr(e.args, '__len__') and len(e.args) >= 1
```

Comment:
Noob question: What would be the cause of this Error with `__len__`? Someone calling `len(dp)` in the middle of iteration?

Reply:
The extra logic here is not related to the DataPipe,but rather, the exception being raised.

It is checking if `e.args` (arguments of the exception) has `__len__` and if it does, there is more than one element. I have seen cases where `e.args` does not mean that criteria.

Reply:
I see. Thanks for the explanation.

Reply:
Your question prompted me to take another look and handle the case where `len(e.args) == 0` when there is no exception message. Have a look.

Reply:
```python
class MyDataPipe(IterDataPipe):

    def __init__(self, x):
        self.x = x

    def __iter__(self):
        raise RuntimeError
        yield 0


dp = MyDataPipe(1)
it = iter(dp)
next(it)
```

Previously, it will just show:
```
Traceback (most recent call last):
  File "/Users/scratch/graphvisualization.py", line 27, in <module>
    next(it)
  File "/Users/pytorch/torch/utils/data/datapipes/_typing.py", line 514, in wrap_generator
    response = gen.send(None)
  File "/Users/scratch/graphvisualization.py", line 21, in __iter__
    raise RuntimeError
RuntimeError: 
```
Now it adds the extra message in the end:
```
This exception is thrown by __iter__ of MyDataPipe(x=1)
```

Reply:
Makes sense. I hope at least TorchData doesn't have such kind of Error without message. lol